runtime-modules/project-token/src/tests/patronage.rs

```diff
@@ -264,271 +173,244 @@ fn decreasing_patronage_rate_fails_with_target_rate_exceeding_current_rate() {
-        .build();
-
-    build_test_externalities(config).execute_with(|| {
-        increase_account_balance(
-            &owner_account,
-            DEFAULT_SPLIT_REVENUE + ExistentialDeposit::get(),
-        );
+    build_default_test_externalities_with_balances(vec![(
+        DEFAULT_ISSUER_ACCOUNT_ID,
+        DEFAULT_SPLIT_REVENUE + ExistentialDeposit::get(),
+    )])
+    .execute_with(|| {
+        IssueTokenFixture::default()
+            .with_patronage_rate(DEFAULT_MAX_YEARLY_PATRONAGE_RATE.into())
+            .execute_call()
+            .unwrap();
+
         IssueRevenueSplitFixture::default().execute_call().unwrap(); // activate revenue split
-        increase_block_number_by(blocks);
+        increase_block_number_by(MIN_REVENUE_SPLIT_TIME_TO_START - 1);
+        let result = ClaimPatronageCreditFixture::default().execute_call();
 
         // expect it to fail even though the staking period is not started yet
-        assert_noop!(
-            Token::claim_patronage_credit(token_id, owner_id),
+        assert_err!(
+            result,
             Error::<Test>::CannotModifySupplyWhenRevenueSplitsAreActive,
         );
     })
 }
 
 #[test]
 fn claim_patronage_ok() {
-    let token_id = token!(1);
-    let owner_id = member!(1).0;
-    let (rate, blocks) = (rate!(10), block!(10));
-
-    let params = TokenDataBuilder::new_empty().with_patronage_rate(rate);
-
-    let config = GenesisConfigBuilder::new_empty()
-        .with_token(token_id, params.build())
-        .with_account(owner_id, ConfigAccountData::default())
-        .build();
+    build_default_test_externalities().execute_with(|| {
+        IssueTokenFixture::default()
+            .with_patronage_rate(DEFAULT_MAX_YEARLY_PATRONAGE_RATE.into())
+            .execute_call()
+            .unwrap();
 
-    build_test_externalities(config).execute_with(|| {
-        increase_block_number_by(blocks);
+        increase_block_number_by(DEFAULT_BLOCK_INTERVAL);
 
-        let result = Token::claim_patronage_credit(token_id, owner_id);
+        let result = ReducePatronageRateToFixture::default().execute_call();
 
         assert_ok!(result);
     })
 }
 
 #[test]
-fn claim_patronage_ok_with_patronage_rate_for_period_capped_at_100pct() {
-    // Simulate condition where patronage_rate.for_period(blocks) > 100%
-    let token_id = token!(1);
-    let owner_id = member!(1).0;
-    let supply = balance!(100);
-    let rate = rate!(55);
-    let blocks = block!(2); // patronage rate for period = 110% > 100%
-
-    let params = TokenDataBuilder::new_empty().with_patronage_rate(rate);
-    // rate for period = 110% but effective rate is capped at 100% -> 100% supply + supply
-    let expected = balance!(100) + supply;
-
-    let config = GenesisConfigBuilder::new_empty()
-        .with_token_and_owner(token_id, params.build(), owner_id, supply)
-        .build();
+fn claim_patronage_ok_with_correct_credit_accounting_and_more_than_100_percent_supply() {
```

Comment:
This also accounts for numeric precision, however there are several tests in the `common/numerical.rs` library for this